### PR TITLE
Make beam-examples run with spark runner

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @amitsela

File: examples/java/src/test/java/org/apache/beam/examples/TfIdfIT.java [hunk 1]
```diff
@@ -16,34 +16,41 @@
...
+@RunWith(JUnit4.class)
+public class TfIdfIT {
 
   @Test
-  public void testTfIdf() throws Exception {
-    SparkPipelineOptions opts = PipelineOptionsFactory.as(SparkPipelineOptions.class);
-    opts.setRunner(SparkRunner.class);
-    Pipeline pipeline = Pipeline.create(opts);
+  public void testE2ETfIdfSpark() throws Exception {
+    SparkPipelineOptions options = PipelineOptionsFactory.as(SparkPipelineOptions.class);
```
Comment (amitsela) [inline examples/java/src/test/java/org/apache/beam/examples/TfIdfIT.java]: Could you use `PipelineOptionsFactory.fromArgs` to be runner agnostic ? Is there a benefit in that ? maybe apply the same to all runners ?

**Comment by amitsela** [comment]
I would still duplicate a WordCount copy into the Spark runner like I did in https://github.com/apache/incubator-beam/pull/539 because it's widely used in the runner's unit tests.
Maybe this could be removed after the runner is mature enough to rely only on the RunnableOnService tests.
And like I also said in https://github.com/apache/incubator-beam/pull/539, transitive dependency is your enemy here, I can't come up with something better than adding Spark provided/runtime dependencies.

This could be resolved by removing the provided scope on spark dependencies from the Spark runner, but I don't think that's a good idea. Looping in @jbonofre WDYT ? this could make the Spark runner Jar become very heavy.. and what about different Spark distributions on clusters ?

**Comment by peihe** [comment]
add R: @davorbonaci

**Comment by amitsela** [comment]
@peihe anything new here ? because https://github.com/apache/incubator-beam/pull/539 is passing tests now - but like you said, it doesn't eliminate code duplication.

I don't see this working if the runner doesn't have a `compile` scope dependency on the engine, and at least for Spark, I'm not sure it's the best way to go.

Pinging @jbonofre: from your experience with customers, is Spark usually provided ?

**Comment by amitsela** [comment]
While my point of view on things is of a Spark (+YARN) cluster, I'm starting to get the feeling that there are a lot of interest in "out-of-the-box" packaging..

Let me raise that in the mailing list to get people's thought on this, and I might change the build to either compile or use profiles or something.

**Pull request closed without merging**